src/clib/lib/enkf/enkf_node.cpp

```diff
@@ -419,11 +433,21 @@ void enkf_node_deserialize(enkf_node_type *enkf_node, enkf_fs_type *fs,
 */
 bool enkf_node_initialize(enkf_node_type *enkf_node, int iens) {
     if (enkf_node->initialize != NULL) {
-        char *init_file =
-            enkf_config_node_alloc_initfile(enkf_node->config, NULL, iens);
-        bool init = enkf_node->initialize(enkf_node->data, iens, init_file);
-        free(init_file);
-        return init;
+        std::unique_ptr<char> init_file(
+            enkf_config_node_alloc_initfile(enkf_node->config, NULL, iens));
+        try {
+            bool init =
+                enkf_node->initialize(enkf_node->data, iens, init_file.get());
```

Comment:
This could be just `return enkf_node->initialize(enkf_node->data, iens, init_file.get());` now.

Reply:
nooooooo we add like tons of extra useful information in the catch block - it's fine to see what file caused the error, but the node type tells you what keyword in your config file you need to find, and the node name gives you the second token in the config file line! superbly useful for finding where to fix it in the config file. i feel. would you agree?